ui/src/app/views/nih-callback/component.tsx

```diff
@@ -0,0 +1,91 @@
+import {Component} from '@angular/core';
+import {Spinner} from 'app/components/spinners';
+import {profileApi} from 'app/services/swagger-fetch-clients';
+import {reactStyles, ReactWrapperBase, withStyle} from 'app/utils';
+import {navigateByUrl} from 'app/utils/navigation';
+
+import * as React from 'react';
+
+const styles = reactStyles({
+  overlay: {
+    backgroundColor: 'transparent',
+    position: 'absolute', top: 0, left: 0, bottom: 0, right: 0,
+    display: 'flex', justifyContent: 'center', alignItems: 'center'
+  },
+  square: {
+    display: 'flex', backgroundColor: 'transparent', borderRadius: 4, padding: '0.5rem'
+  },
+  error: {
+    fontWeight: 600,
+    color: '#2F2E7E',
+    margin: '1rem'
+  }
+});
+
+export const Error = withStyle(styles.error)('div');
+const spinner = <div style={styles.overlay}><div style={styles.square}><Spinner /></div></div>;
+
+export interface NihCallbackState {
+  error: boolean;
+  errorMessage: string;
+}
+
+export class NihCallback extends React.Component<{}, NihCallbackState> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: false,
+      errorMessage: ''
+    };
+  }
+
+  async componentDidMount() {
+    // Assumes callback url has format of `/nih-callback?token=XYZ`
+    const token = window.location.search.replace('?token=', '');
```

Comment:
This is brittle as it assumes no other query parameters. Should use a library to extract the query params instead, e.g.

```
(new URL(window.location)).searchParams.get('token');
```

Reply:
Got it - will do.

Reply:
Done